CRM/Contribute/Form/ContributionBase.php

```diff
@@ -1242,6 +1243,48 @@ protected function getExistingContributionID(): ?int {
     return $this->_ccid ?: CRM_Utils_Request::retrieve('ccid', 'Positive', $this);
   }
 
+  /**
+   * Get the submitted value, accessing it from whatever form in the flow it is
+   * submitted on.
+   *
+   * @param string $fieldName
+   *
+   * @return mixed|null
+   */
+  public function getSubmittedValue(string $fieldName) {
+    $value = $this->controller->exportValue('Main', $fieldName);
+    if (in_array($fieldName, $this->submittableMoneyFields, TRUE)) {
+      return CRM_Utils_Rule::cleanMoney($value);
+    }
+
+    // Numeric fields are not in submittableMoneyFields (for now)
+    $fieldRules = $this->_rules[$fieldName] ?? [];
```

Comment:
what is _rules here?

Reply:
@seamuslee001 yeah - that part would only work on the Main page really - it doesn't matter for the amount implemented so far but I'll tinker with it more in master